src/object/srv_obj_migrate.c

```diff
@@ -2214,7 +2214,8 @@ destroy_existing_container(struct migrate_pool_tls *tls, uuid_t cont_uuid)
 				DP_RC(rc));
 		}
 
-		rc = ds_cont_tgt_destroy(tls->mpt_pool_uuid, cont_uuid);
+		rc = ds_cont_tgt_destroy_this_xstream(tls->mpt_pool_uuid,
+						      cont_uuid);
```

Comment:
You can not call destroy_this_xstream here. I think this is still on xstream 0. so it does need collective call to destroy container on all xstreams.

I thought the original sulotion is correct, but you need somehow skip the cont_delete_ec_agg() for this "destory", which is only needed for normal container destory.